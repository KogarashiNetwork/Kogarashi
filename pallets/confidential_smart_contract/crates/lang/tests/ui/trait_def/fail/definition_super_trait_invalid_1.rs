use ink_lang as ink;

pub trait SuperTraitDefinition {}

#[ink::trait_definition]
pub trait TraitDefinition: SuperTraitDefinition {
    #[ink(message)]
    fn message(&self);
}

fn main() {}
