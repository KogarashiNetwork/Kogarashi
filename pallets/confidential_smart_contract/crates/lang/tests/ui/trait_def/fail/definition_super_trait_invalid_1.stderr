error: ink! trait definitions with supertraits are not supported, yet
 --> tests/ui/trait_def/fail/definition_super_trait_invalid_1.rs:6:28
  |
6 | pub trait TraitDefinition: SuperTraitDefinition {
  |                            ^^^^^^^^^^^^^^^^^^^^
